packages/opentelemetry/lib/tracing/tracerProvider.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable no-unused-vars */
+
+import { getBatchSpanProcessor } from './spanProcessor';
+import { buildExporter, AVAILABLE_EXPORTERS } from './exporter';
+import { NodeTracerProvider } from '@opentelemetry/node';
+import { registerInstrumentations, InstrumentationOption } from '@opentelemetry/instrumentation';
+import { ServerInstrumentation } from './serverInstrumenation';
+import { diag, DiagConsoleLogger, DiagLogLevel } from '@opentelemetry/api';
+import { SpanProcessor, } from '@opentelemetry/tracing';
+
+//Delegate
+class TracerProvider {
+  constructor () {
+    this.provider = new NodeTracerProvider();
+    this.exporters = {};
+    this.init();
+  }
+
+  init () {
+    diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.ALL);
+    this.provider.register();
+
+    const consoleExporter = buildExporter(AVAILABLE_EXPORTERS.CONSOLE);
+    const spanProcessor = getBatchSpanProcessor(consoleExporter);
+    this.addSpanProcessor(spanProcessor);
+    this.exporters[AVAILABLE_EXPORTERS.CONSOLE] = consoleExporter;
+
+    this._serverInstrumentation = new ServerInstrumentation();
+    registerInstrumentations({
+      instrumentation: this._serverInstrumentation.instrumentations,
+      tracerProvider: this.provider
+    });
+    this._active = true;
+    this.currentConfig = {
+      active: this._active,
+      current_exporters: [AVAILABLE_EXPORTERS.CONSOLE],
+      exporters:
+        [
+          {
+            exporter_type: 'console',
+            config: {}
+          }
+        ]
+    };
+  }
+
+  getCurrentConfig () {
+    return JSON.parse(JSON.stringify(this.currentConfig));
+  }
+
+  /**
+   * adds exporter type and config to currentConfig
+   * @param {[Object]} exporter  [exporter object with exporter_type and config]
+   */
+  addExporterToConfig (exporter) {
+    this.currentConfig.current_exporters.push(exporter.exporter_type);
+    this.currentConfig.exporters.push(exporter);
+  }
+
+  /**
+   * generates a span processor for exporter object containing a type and config
+   * @param {[Object]} exporter  [exporter object with exporter_type and config]
```

Comment:
I would rather use typefed to define which properties this object has. See https://github.com/appium/appium-base-driver/blob/cfb657b29e810dd53eaa2ae23a1e7d847246b834/lib/basedriver/image-element.js#L18 for an example